### PR TITLE
Add removeByGidAsync() to spi.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

**Review by geirst — DISMISSED** [review]
👍

File: persistence/src/vespa/persistence/spi/doctype_gid_and_timestamp.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    DocTypeGidAndTimestamp();
+    DocTypeGidAndTimestamp(const vespalib::string& doc_type_, document::GlobalId gid_, Timestamp timestamp_) noexcept;
+
+    DocTypeGidAndTimestamp(const DocTypeGidAndTimestamp&);
+    DocTypeGidAndTimestamp& operator=(const DocTypeGidAndTimestamp&);
+    DocTypeGidAndTimestamp(DocTypeGidAndTimestamp&&) noexcept;
+    DocTypeGidAndTimestamp& operator=(DocTypeGidAndTimestamp&&) noexcept;
+
+    bool operator==(const DocTypeGidAndTimestamp& rhs) const noexcept {
+        return ((doc_type == rhs.doc_type) && (gid == rhs.gid));
```
[the remark this answers is not in the document]
Comment (toregge) [reply persistence/src/vespa/persistence/spi/doctype_gid_and_timestamp.h]: No.

File: persistence/src/vespa/persistence/spi/doctype_gid_and_timestamp.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    void print(vespalib::asciistream&) const;
+    vespalib::string to_string() const;
+
+    struct hash {
+        size_t operator()(const DocTypeGidAndTimestamp& dt_gid_ts) const noexcept {
+            const size_t h = document::GlobalId::hash()(dt_gid_ts.gid);
+            return h ^ (dt_gid_ts.timestamp + 0x9e3779b9U + (h << 6U) + (h >> 2U)); // Basically boost::hash_combine
```
[the remark this answers is not in the document]
Comment (toregge) [reply persistence/src/vespa/persistence/spi/doctype_gid_and_timestamp.h:45]: No.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**